gen3-client/g3cmd/upload.go

```diff
@@ -36,41 +40,110 @@ func init() {
 			function.Config = configure
 			function.Request = request
 
-			endPointPostfix := "/user/data/upload/" + guid
-
-			respURL, err := function.DoRequestWithSignedHeader(profile, "", endPointPostfix)
+			filePaths, err := filepath.Glob(uploadPath)
```

Comment:
this funciton matches anything with regex, which means it only match the folder and the process just panic if user specify `dir/` as you had in the example.

Reply:
will update the example to make it more clear and provide error if it is a folder